### PR TITLE
Background add-in and ribbon APIs

### PR DESCRIPTION
**Pull request by sumurthy**
We are releasing background addins API and Office ribbon APIs. This definition adds the API details. 

Please fill in this template.
- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/office-js-preview/index.d.ts
Comment (Rick-Kirkham) [inline types/office-js-preview/index.d.ts]:
```suggestion
         * Collection of tabs whose state is set with the call of `requestUpdate`.
```

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Can we rephrase this to describe what's happening rather than repeating the enum name? Like, this is the behavior of the add-in when the Office host starts or when the add-in is opened?

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Grammar consistency: "Check" -> "Checks"

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Can we be more descriptive here? Maybe "The add-in does not load until opened by the user."

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: "the event" - which event is that?

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Later, you use "VisbilityModeChanged event" to describe this. I like using the actual name. Can we surround it with code ticks (`)?

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Can we surround VisibilityModeChanged with code ticks (`)?

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Shouldn't this method be `hideTaskpane`?

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Why is this `showAsTaskpane` and not `showTaskpane`?

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: TSDoc prefers a hyphen between the parameter name and the description.
https://github.com/Microsoft/tsdoc

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Grammar nit: It sounds a little awkward to say "Ribbon interface contains". How about "An interface that contains"?
That way, we can avoid having to worry about denoting the interface name with code ticks too.

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Typo: "request for an update" -> "request an update"

**Review by AlexJerabek — CHANGES_REQUESTED** [review]

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: The UI in the future could be shown as dialog or taskpane. for now only taskpane is supported. So, we'd like to have a generic hide API.

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: It's a generic type and can be used across any events that returns this function.
this is to avoid having to need the object to call the event off method.

Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts]: Then shouldn't it be "an event" instead of "the event"?

**Review by Rick-Kirkham — APPROVED** [review]

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: It is in a way add-ins runtime settings. We wanted to avoid using the work runtime. 
Using this object you can control addins display, startup behavior, etc.

Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts]: Can we add that sort of information to the description?

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: done

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: Done

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: Done

File: types/office-js-preview/index.d.ts
Comment (sumurthy) [reply types/office-js-preview/index.d.ts]: Done

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: "UI is Hidden" -> "UI is hidden."

File: types/office-js-preview/index.d.ts
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Please add a period to the end of this description.

**Review by AlexJerabek — APPROVED** [review]
Approving! Thanks for making those changes. There's a couple minor things that do not need a re-review.

**Comment by Rick-Kirkham** [comment]
PLEASE WAIT FOR SECOND REVIEWER APPROVAL BEFORE MERGING.

**Comment by AlexJerabek** [comment]
@Rick-Kirkham, in addition to my review?

**Comment by Rick-Kirkham** [comment]
> @Rick-Kirkham, in addition to my review?

My bad. I didn't realize that you'd already approved. I deleted my comment.

**Pull request merged**
